elytron-oidc-client/src/test/java/org/wildfly/extension/elytron/oidc/ElytronOidcClientSubsystemTransformerTestCase.java

```diff
@@ -0,0 +1,79 @@
+        return List.<Object[]>of(
+                new Object[] { ModelTestControllerVersion.EAP_8_0_0, ElytronOidcClientSubsystemModel.VERSION_2_0_0.getVersion() }
+        );
+    }
+    private final ModelTestControllerVersion controllerVersion;
+    private final ModelVersion modelVersion;
+
+    public ElytronOidcClientSubsystemTransformerTestCase(ModelTestControllerVersion controllerVersion, ModelVersion version) {
+        super(ElytronOidcExtension.SUBSYSTEM_NAME, new ElytronOidcExtension(), ElytronOidcSubsystemSchema.VERSION_2_0_PREVIEW, ElytronOidcSubsystemSchema.CURRENT.get(Stability.PREVIEW));
+        this.controllerVersion = controllerVersion;
+        this.modelVersion = version;
+    }
+
+    @Test
+    public void testTransformations() throws Exception {
+        final ModelVersion version = modelVersion;
+        KernelServices services = this.buildKernelServices(controllerVersion, version);
+
+        checkSubsystemModelTransformation(services, version, null, false);
+        ModelNode transformed = services.readTransformedModel(version);
+        Assert.assertTrue(transformed.isDefined());
+    }
+
+    private KernelServices buildKernelServices(ModelTestControllerVersion controllerVersion, ModelVersion version) throws Exception {
+        KernelServicesBuilder builder = this.createKernelServicesBuilder(new DefaultInitializer(this.getSubsystemSchema().getStability()))
+                .setSubsystemXmlResource("elytron-oidc-client-transform.xml");
+        builder.createLegacyKernelServicesBuilder(AdditionalInitialization.ADMIN_ONLY_HC, controllerVersion, version)
+                .addMavenResourceURL(String.format("%s:wildfly-elytron-oidc-client-subsystem:%s", controllerVersion.getMavenGroupId(), controllerVersion.getMavenGavVersion()))
+                .skipReverseControllerCheck()
+                .addParentFirstClassPattern("org.jboss.as.controller.logging.ControllerLogger*")
+                .addParentFirstClassPattern("org.jboss.as.controller.PathAddress")
+                .addParentFirstClassPattern("org.jboss.as.controller.PathElement")
+                .addParentFirstClassPattern("org.jboss.as.server.logging.*")
+                .addParentFirstClassPattern("org.jboss.logging.*")
+                .addParentFirstClassPattern("org.jboss.dmr.*")
+                .dontPersistXml();
+
+        KernelServices services = builder.build();
+        Assert.assertTrue(ModelTestControllerVersion.MASTER + " boot failed", services.isSuccessfulBoot());
+        Assert.assertTrue(controllerVersion.getMavenGavVersion() + " boot failed", services.getLegacyServices(version).isSuccessfulBoot());
+
+        return services;
+    }
+
+    @Override
+    protected void compareXml(String configId, String original, String marshalled) {
+        //
+    }
+
+    @Override
+    protected String getSubsystemXml() throws IOException {
+        return readResource("elytron-oidc-client-transform.xml");
```

Comment:
I also see an `elytron-oidc-client-reject.xml` file but it seems to be empty. Is that getting used anywhere?

Reply:
It should be there now. I hadn't written a rejection test before the last push. 
Thanks!